pkg/config/config.go

```diff
@@ -64,8 +67,16 @@ func FromYAMLFile(name string) (*Config, error) {
 		return nil, errors.Wrap(err, "can't parse YAML file "+name)
 	}
 
+	// Decode again with yaml.DisallowUnknownField() (like v1.2 will do) and issue a warning if it returns an error.
+	c.DecodeWarning = yaml.NewDecoder(bytes.NewReader(f), yaml.DisallowUnknownField()).Decode(&Config{})
+
 	if err := c.Validate(); err != nil {
-		return nil, errors.Wrap(err, "invalid configuration")
+		msg := "invalid configuration"
+		if warn := c.DecodeWarning; warn != nil {
+			return nil, fmt.Errorf("%s: %w\n\nwarning: ignored unknown config option:\n\n%v", msg, err, warn)
+		} else {
+			return nil, errors.Wrap(err, msg)
+		}
```

Comment:
I doubt we need this. After all it failed because of _err_, not _warn_.

Reply:
I think it can be quite useful, for example if things are missing because you've misnamed them:

```
invalid configuration: database host missing

warning: ignored unknown config option:

[3:3] unknown field "hostname"
       1 | database:
       2 |   type: mysql
    >  3 |   hostname: localhost
             ^
       4 |   port: 3306
       5 |   database: icingadb
       6 |   user: icingadb
       7 |   
```

Reply:
Ack.